non-linear-solver-spec.json

```diff
@@ -319,5 +322,23 @@
         "default": 100,
         "type": "int",
         "doc": "Dangerous Option: Quit the optimization if the solver reduces the energy by less than f_delta for consecutive f_delta_step_tol steps."
+    },
+    {
+        "pointer": "/advanced/gradient_fd",
```

Comment:
maybe this instead of a bool is a string with 3 opts? none, full and directional?